### PR TITLE
[BUG] Making crlibm an optional dependency

### PR DESCRIPTION
**Pull request by naoise-h**
The `crlibm` library is presenting difficulties in installing on Windows (see #66), and, as a required dependency, is preventing the successful installation of diffprivlib. As only one function of `crlibm` is used in only one instance in diffprivlib (in the `Snapping` mechanism), it is proposed through this PR to make `crlibm` an optional dependency. In the absence of `crlibm`, the corresponding function from `numpy` will be used instead.

The `crlibm` library can be installed during installation of diffprivlib by specifying the `crlibm` option as follows: `pip install diffprivlib[crlibm]`.

### CODE REVIEW

File: diffprivlib/mechanisms/snapping.py [hunk 1]
```diff
@@ -4,8 +4,11 @@
 """
 import struct
 
-import crlibm
 import numpy as np
+try:
+    from crlibm import log_rn
+except ModuleNotFoundError:
+    log_rn = np.log
```
Comment (stefano81) [inline diffprivlib/mechanisms/snapping.py:11]: A info level log message would be useful

**Review by stefano81 — APPROVED** [review]

**Review by stefano81 — APPROVED** [review]
LGTM, beside CodeCov warning

File: diffprivlib/mechanisms/snapping.py (hunk 1 above)
Comment (naoise-h) [reply diffprivlib/mechanisms/snapping.py:11]: Thanks, I'll keep this comment on the todo list for the future.

**Comment by naoise-h** [comment]
> LGTM, beside CodeCov warning

I believe the CodeCov comparison is with an older commit (we have since switched from Travis to Github actions), so the diff is not fully relevant to this PR. That's the only way to explain the extra 18 lines of code that CodeCov now hits, as far as I know...

**Comment by stefano81** [comment]
I would suggest to disable CodeCov if it is not assessing the current commit

**Pull request merged**